include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -352,6 +352,21 @@ def SubtagOp : FIRRTLExprOp<"subtag"> {
   }];
 }
 
+def TagExtractOp : FIRRTLExprOp<"tagof", [InferTypeOpInterface]> {
```

Comment:
just an opinion, I feel like it reduces the mental overhead to match mnemonic to the op name. Suggestion being `ExtractTagOp` and `extract_tag` or `TagOfOp` and `tag_of`. (not sure about our naming conventions regarding the `_`s)